### ISSUE TITLE
Arrays containing big-endian numbers are wrongly decoded

### ISSUE THREAD

**Issue by loichuder** [usnistgov/h5wasm#49]
When reading a dataset containing values (floats or integers) stored with big endianness, the resulting values are wrong.

I guess it comes from the  `process_data` function that only expects data with little endianness ?

It is very easy to reproduce by creating a file like this and reading it with h5wasm:

```python
import h5py
import numpy as np

with h5py.File("BE.h5", "w") as f:

    # Note the `>` in `dtype` that tells that this is big-endian
    f.create_dataset("float", data=np.arange(0.01, 0.11, 0.01), dtype=">f8")
    f.create_dataset(
        "int", data=np.arange(8).reshape(2, 2, 2), dtype=">i8", shape=(2, 2, 2)
    )
```

Related to https://github.com/silx-kit/h5web/issues/1421#issuecomment-1513061952

**Comment by bmaranville** [comment]
should be fixed by v0.4.11, published to npm and github just now.

**Comment by loichuder** [comment]
It indeed solves our issue ! 

Thanks a lot for the quick fix, this is greatly appreciated :slightly_smiling_face: !
